codex-cli/src/components/chat/terminal-chat.tsx

```diff
@@ -275,6 +278,49 @@ export default function TerminalChat({
     };
   }, [loading, confirmationPrompt]);
 
+  // Notify desktop with a preview when an assistant response arrives
+  const prevLoadingRef = useRef<boolean>(false);
+  useEffect(() => {
+    // Only notify when notifications are enabled
+    if (!notify) {
+      prevLoadingRef.current = loading;
+      return;
+    }
+    if (
+      prevLoadingRef.current &&
+      !loading &&
+      confirmationPrompt == null &&
+      items.length > 0
+    ) {
+      if (process.platform === "darwin") {
+        // find the last assistant message
+        const assistantMessages = items.filter(
+          (i) => i.type === "message" && i.role === "assistant",
+        );
+        const last = assistantMessages[assistantMessages.length - 1];
+        if (last) {
+          const text = last.content
+            .map((c) => {
+              if (c.type === "output_text") {
+                return c.text;
+              }
+              return "";
+            })
+            .join("")
+            .trim();
+          const preview = text.replace(/\n/g, " ").slice(0, 100);
+          const safePreview = preview.replace(/"/g, '\\"');
+          const title = "Codex CLI";
+          const cwd = PWD;
+          exec(
```

Comment:
Second, we should absolutely not be using `exec()` with a single string arg that appears to be subject to injection risks. At a minimum, we should be using `spawn()` with a list of strings.